sql/datetimetype.go

```diff
@@ -273,9 +273,6 @@ func (t datetimeType) ConvertWithoutRangeCheck(v interface{}) (time.Time, error)
 		return zeroTime, ErrConvertToSQL.New(t)
 	}
 
-	if t.baseType == sqltypes.Date {
```

Comment:
What is the basis of this change? `DATE` types shouldn't have any component more specific than the day. In MySQL, displaying the contents of a `DATE` column in multiple ways (using the `CONVERT` function or altering the table's schema to be `DATETIME` instead of `DATE`) always displays the HH:MM:SS+ portion as all zeros. This change doesn't feel correct.

Reply:
I see, I think in the second MySQL case it's using the widest type as the comparator, and I _think_ we do the same but `DATE` does not widen itself to `DATETIME`, and I think it probably should. If you change `/sql/datetimetype.go::Promote()` to
```go
// Promote implements the Type interface.
func (t datetimeType) Promote() Type {
	return Datetime
}
```
Does GMS match the behavior of the 2nd query? If so, I'd rather make that change than removing the truncation. If not, then we can keep the truncation removal (even though I'm still not happy about it lol).

Reply:
this works! much better than rewriting all those tests